### PR TITLE
#409 - Added interface checking on object types

### PR DESCRIPTION
**Pull request by bbakerman**

### CODE REVIEW

File: src/main/java/graphql/schema/validation/ValidationRule.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package graphql.schema.validation;
 
 import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLType;
 
 public interface ValidationRule {
```
Comment (andimarek) [inline src/main/java/graphql/schema/validation/ValidationRule.java]: We should rename this to `SchemaValidationRule` or something similar ... `ValidationRule` too generic

File: src/main/java/graphql/schema/validation/Validator.java [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
+
+        List<GraphQLType> types = schema.getAllTypesAsList();
+        types.forEach(type -> {
+            for (ValidationRule rule : rules) {
+                rule.check(type, validationErrorCollector);
+            }
+        });
+
         traverse(schema.getQueryType(), rules, validationErrorCollector);
         if (schema.isSupportingMutations()) {
```
Comment (andimarek) [inline src/main/java/graphql/schema/validation/Validator.java]: here I think `subscription` is missing?

Comment (bbakerman) [inline src/main/java/graphql/schema/validation/Validator.java]: yes it is missing that.  We can do that in another PR.  

https://github.com/graphql-java/graphql-java/issues/414
